### ISSUE TITLE
New Rule: UseOrEmpty

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#3465]
https://twitter.com/kotlin/status/1337306632477356034

Noncompliant Code:
```kotlin
fun test(x: List<Int>?) {
    val y = x ?: emptyList()
}
```

Compliant Code:
```kotlin
fun test(x: List<Int>?) {
    val y = x.orEmpty()
}
```

**Comment by BraisGabin** [comment]
I agree with this rule. The second code is better. But, for me, both of them are code smells. If you don't want to handle null lists in your function you should mark the parameter as not null. Then is work of the caller to make the null list empty if they want to call it. This is usually recursive and you end up with less null values.

But I don't know how could we create a rule to check this without LOTS of false positives...

**Comment by cortinico** [comment]
I second the rule. I also don't think there would be too many false positives. Imho a rule that checks for every `?: emptyList()` and returns a report would just work fine 🤔

**Comment by BraisGabin** [comment]
No, no, this rule is fine. I think that I didn't explain my self properly. What I mean is that if you need to use `?: emptyList` or `orEmpty` on a list passed as parameter in a function, then the smell code is in the parameter itself. It should not be nullable in the first place.
